flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/OceanBaseConnection.java

```diff
@@ -49,7 +67,111 @@ public static Configuration config(
                 .build();
     }
 
-    public static JdbcConnection.ConnectionFactory factory(ClassLoader classLoader) {
-        return JdbcConnection.patternBasedFactory(URL_PATTERN, DRIVER_CLASS_NAME, classLoader);
+    private static String formatJdbcUrl(String jdbcDriver, Properties jdbcProperties) {
+        Properties combinedProperties = new Properties();
+        combinedProperties.putAll(DEFAULT_JDBC_PROPERTIES);
+        if (jdbcProperties != null) {
+            combinedProperties.putAll(jdbcProperties);
+        }
+        String urlPattern =
+                jdbcDriver.toLowerCase().contains("oceanbase") ? OB_URL_PATTERN : MYSQL_URL_PATTERN;
```

Comment:
If `compatibleMode` could be decided in the constructor, this condition could be like `compatibleMode.equals(xxx)`.